Fix deprecation warnings

When using lts-11.13, we get the following deprecation warnings
```
saml2-web-sso/src/SAML2/WebSSO/API/Example.hs:19:1: warning: [-Wdeprecations]
    Module ‘Servant.Utils.Enter’ is deprecated:
      Use hoistServer or hoistServerWithContext from servant-server
   |
19 | import Servant.Utils.Enter
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

saml2-web-sso/src/SAML2/WebSSO/API/Example.hs:32:11: warning: [-Wdeprecations]
    In the use of type constructor or class ‘Enter’
    (imported from Servant.Utils.Enter):
    Deprecated: "Use hoistServer or hoistServerWithContext from servant-server"
   |
32 |         ( Enter (ServerT APPAPI m) m Handler (Server APPAPI)
   |           ^^^^^

saml2-web-sso/src/SAML2/WebSSO/API/Example.hs:37:39: warning: [-Wdeprecations]
    In the use of ‘enter’ (imported from Servant.Utils.Enter):
    Deprecated: "Use hoistServer or hoistServerWithContext from servant-server"
   |
37 |       served = serve (Proxy @APPAPI) (enter (NT (nt @SimpleError @m ctx)) appapi :: Server APPAPI)
   |                                       ^^^^^
```

--- a/src/SAML2/WebSSO/API/Example.hs
+++ b/src/SAML2/WebSSO/API/Example.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP               #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 {-# OPTIONS_GHC -Wno-orphans #-}
@@ -16,6 +17,9 @@ import Network.Wai hiding (Response)
 import SAML2.WebSSO
 import Servant.API hiding (URI(..))
 import Servant.Server
+#if !MIN_VERSION_servant_server(0,12,0)
+import Servant.Utils.Enter
+#endif
 import Text.Hamlet.XML
 import Text.XML
 import Text.XML.Util
@@ -28,11 +32,20 @@ app :: IO Application
 app = app' (Proxy @SimpleSP) =<< mkSimpleSPCtx =<< configIO
 
 app' :: forall (m :: * -> *).
-        ( SP m, SPHandler SimpleError m
+        (
+#if !MIN_VERSION_servant_server(0,12,0)
+          Enter (ServerT APPAPI m) m Handler (Server APPAPI),
+#endif
+          SP m, SPHandler SimpleError m
         ) => Proxy m -> NTCTX m -> IO Application
 app' Proxy ctx = do
   let served :: Application
-      served = serve (Proxy @APPAPI) (hoistServer (Proxy @APPAPI) (nt @SimpleError @m ctx) appapi :: Server APPAPI)
+      served = serve (Proxy @APPAPI)
+#if MIN_VERSION_servant_server(0,12,0)
+                   (hoistServer (Proxy @APPAPI) (nt @SimpleError @m ctx) appapi :: Server APPAPI)
+#else
+                   (enter (NT (nt @SimpleError @m ctx)) appapi :: Server APPAPI)
+#endif
   pure . setHttpCachePolicy $ served
 
 type SPAPI =

--- a/src/SAML2/WebSSO/API/Example.hs
+++ b/src/SAML2/WebSSO/API/Example.hs
@@ -16,7 +16,6 @@ import Network.Wai hiding (Response)
 import SAML2.WebSSO
 import Servant.API hiding (URI(..))
 import Servant.Server
-import Servant.Utils.Enter
 import Text.Hamlet.XML
 import Text.XML
 import Text.XML.Util
@@ -29,12 +28,11 @@ app :: IO Application
 app = app' (Proxy @SimpleSP) =<< mkSimpleSPCtx =<< configIO
 
 app' :: forall (m :: * -> *).
-        ( Enter (ServerT APPAPI m) m Handler (Server APPAPI)
-        , SP m, SPHandler SimpleError m
+        ( SP m, SPHandler SimpleError m
         ) => Proxy m -> NTCTX m -> IO Application
 app' Proxy ctx = do
   let served :: Application
-      served = serve (Proxy @APPAPI) (enter (NT (nt @SimpleError @m ctx)) appapi :: Server APPAPI)
+      served = serve (Proxy @APPAPI) (hoistServer (Proxy @APPAPI) (nt @SimpleError @m ctx) appapi :: Server APPAPI)
   pure . setHttpCachePolicy $ served
 
 type SPAPI =

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,4 +1,4 @@
-resolver: lts-10.3
+resolver: lts-11.13
 
 packages:
 - .

--- a/test/Test/SAML2/WebSSO/APISpec.hs
+++ b/test/Test/SAML2/WebSSO/APISpec.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP               #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 {-# OPTIONS_GHC -Wno-unused-binds -Wno-incomplete-patterns -Wno-incomplete-uni-patterns -Wno-orphans #-}
@@ -14,7 +15,9 @@ import Data.String.Conversions
 import Lens.Micro
 import SAML2.WebSSO
 import Servant
+#if !MIN_VERSION_servant_server(0,12,0)
 import Servant.Utils.Enter
+#endif
 import Shelly (shelly, run, setStdin, silently)
 import Test.Hspec hiding (pending)
 import Test.Hspec.Wai
@@ -51,10 +54,19 @@ base64theirs sbs = shelly . silently $ cs <$> (setStdin (cs sbs) >> run "/usr/bi
 -- on servant-0.12 or later, use 'hoistServer':
 -- <https://github.com/haskell-servant/servant/blob/master/servant/CHANGELOG.md#significant-changes-1>
 withapp :: forall (api :: *).
-  ( Enter (ServerT api TestSP) TestSP Handler (Server api)
-  , HasServer api '[]
+  (
+#if !MIN_VERSION_servant_server(0,12,0)
+    Enter (ServerT api TestSP) TestSP Handler (Server api),
+#endif
+    HasServer api '[]
   ) => Proxy api -> ServerT api TestSP -> IO Ctx -> SpecWith Application -> Spec
-withapp proxy handler mkctx = with (mkctx <&> \ctx -> serve proxy (enter (NT (nt @SimpleError @TestSP ctx)) handler :: Server api))
+withapp proxy handler mkctx = with (mkctx <&> \ctx -> serve proxy
+#if MIN_VERSION_servant_server(0,12,0)
+                                     (hoistServer (Proxy @api) (nt @SimpleError @TestSP ctx) handler :: Server api)
+#else
+                                     (enter (NT (nt @SimpleError @TestSP ctx)) handler :: Server api)
+#endif
+                                   )
 
 
 spec :: Spec

